File: src/cpp/hello_name.cpp
/*
 * C++
 * Asks the user for their name and says hello
 * compile: g++ c-plus-plus.cpp -o c-plus-plus
 */

#include <iostream> // used to get input and output (cin cout)
#include <string> // used for string variables
using namespace std; // using the standard library so we dont have to use ::

int main(void){
	// display prompt
	cout << "Enter your name: ";

	// get user input
	string name = "\0";
	cin >> name;

	// say hello
	cout << "Hello " << name << "!\n";
	
	// exit without error
	return 0;
}

File: src/java/hello_name.java
/*
 * JAVA (Requires Java Runtime Environment (jre)
 * Asks for the users name and says hi
 * compile: javac java.java
*/

import java.util.Scanner; // used for taking user input

public class java{
	public static void main(String args[]){
		// display prompt
		System.out.print("Enter your name: ");
		
		// get users name
		Scanner input = new Scanner(System.in);
		String name;
		name = input.nextLine();

		// say hello
		System.out.println("Hello " + name + "!");
	}
}
